backend/monitor/monitor.go

```diff
@@ -57,6 +59,16 @@ type FailureHandler interface {
 // HandleUpdate causes the Monitor to observe the event. If the monitor has
 // been stopped, this method has no effect.
 func (m *Monitor) HandleUpdate(event *types.Event) error {
+	// If the monitor has not started yet, wait 1 second to avoid a data race on
+	// resetChan and check again.
+	if !m.IsStarted() {
```

Comment:
They are, but what we were seeing was in some cases the resetChan hadn't been initialized by the time HandleUpdate was called. This seems like it shouldn't be the case though as I would think New would only complete after the call to start() completes.

Reply:
I'm having a hard time understanding how `resetChan` could possibly not be initialized by the time `HandleUpdate` is called (since we are doing `m.resetChan = make(chan time.Duration)` directly in `start()`).

I have no doubt this PR could actually fix the keepalive problem we have, I just feel like it we didn't really addressed the root cause (which may be fine too). I can try to take a look once I'm done with my current task!

Reply:
So this data race is still somewhat of a concern to me since the monitor gets initialized and started closely to the goroutine calling `processKeepalives`. However, I do feel like on my working branch I've addressed that root cause, just hammering on the testing to ensure that! I have to close this PR anyway and reopen it for Travis anyway ;)

Reply:
If it's started before processKeepalives, then it's fine.